src/rgw/rgw_admin.cc

```diff
@@ -5560,6 +5560,50 @@ int main(int argc, const char **argv)
   } /* OPT_BUCKETS_LIST */
 
   if (opt_cmd == OPT_BUCKET_STATS) {
+    if (bucket_name.empty() && !bucket_id.empty()) {
+      void *handle;
+      int ret = store->ctl()->meta.mgr->list_keys_init("bucket.instance", marker, &handle);
+      if (ret < 0) {
+        cerr << "ERROR: can't get key: " << cpp_strerror(-ret) << std::endl;
+        return -ret;
+      }
+
+      bool truncated;
+      uint64_t count = 0;
+      string s;
+      string token;
+      string delimiter = ":";
+      uint64_t pos = 0;
+      bool bucket_name_from_bucket_id = false;
+      do {
+        list<string> keys;
+        ret = store->ctl()->meta.mgr->list_keys_next(handle, 1, keys, &truncated);
+        if (ret < 0 && ret != -ENOENT) {
```

Comment:
do you know where this special case for ENOENT came from? if list_keys_next() fails with ENOENT, it probably isn't setting truncated and that could result in an infinite loop. so i think we should just fail out on -ENOENT, the same way we do with any other ret < 0

Reply:
hmm, looks like I got it from here -https://github.com/ceph/ceph/blob/master/src/rgw/rgw_admin.cc#L6910